packages/ui/src/components/LedgerDevice/structure/ledgerActions.ts

```diff
@@ -48,90 +59,128 @@ const isAvailable = async (): Promise<boolean> => {
+        _PATH.pathIndex += 1;
       }
-    }
+    });
   }
+  return foundIndex;
+}
 
-  // Remap of BigInt values from strings creates issues in this cast 
-  // So forcing the two affected fields (amount,assetTotal) back to numeric
-  if ('amount' in txn) {
-    const parsed = parseInt(txn['amount']);
-    // Soft check on the result mating the txn amount because it is an expect int to string compare
-    if (isNaN(parsed) || parsed != txn['amount']) { 
-      errors.push('Ledger transaction amount must be an integer.'); 
-    }
-    else {
-      txn['amount'] = parsed;
-    }
+///
+// Tries to get multiple Ledger accounts
+// Returns an array of publicKey addresses, encoded 
+///
+const getAllAddresses = async (): Promise<LedgerActionResponse> => {
+  const accounts = Array<object>();
+  let errorOnIndex = false;
+  const maxAccounts = 8; // Arbitrary - to prevent infinite loops
+  let lar: LedgerActionResponse = {};
+
+  // Reset path index to get all 
+  _PATH.pathIndex = 0;
+
+  // If we haven't connected yet, do it now. This will prompt the tab to ask for device.
+  if (!ledgerTransport) {
+    ledgerTransport = await getDevice().catch((e) => {
+      // If this is a known error from Ledger it will contain a message
+      lar =
+        e && 'message' in e
+          ? { error: e.message }
+          : { error: 'An unknown error has occured in connecting the Ledger device.' };
+    });
   }
-  if ('assetTotal' in txn) {
-    const parsed = parseInt(txn['assetTotal']);
-    // Soft check on the result mating the txn assetTotal because it is an expect int to string compare
-    if (isNaN(parsed) || parsed != txn['assetTotal']) { 
-      errors.push('Ledger transaction assetTotal must be an integer.'); 
-    }
-    else {
-      txn['assetTotal'] = parsed;
-    }
+
+  // Return error if we have one
+  if (lar.error) {
+    return lar;
+  }
+
+  while(!errorOnIndex && _PATH.pathIndex < maxAccounts) {
+    const currentIndex = `${_PATH.pathIndex}`;
+    await ledgerTransport
+      .getAddress(_PATH.current)
+      .then((o: any) => {
+        const publicAddress: string = algosdk.encodeAddress(Buffer.from(o.publicKey, 'hex'));
+        const retrievedAccount = { 'ledgerIndex': currentIndex, 'hex': o.publicKey, 'publicAddress': publicAddress };
+        accounts.push(retrievedAccount);
+      })
+      .catch((e) => {
+        console.log(e)
+        // Abort on error and pass back the current foundIndex
+        errorOnIndex = true;
+        lar.error = e;
+      })
+      .finally(_PATH.pathIndex += 1);
   }
 
-  const builtTxn = new algosdk.Transaction(txn);
+  lar.message = accounts;
+  return lar;
+}
 
-  if ('group' in txn && txn['group']) {
-    // Remap group field lost from cast
-    builtTxn.group = Buffer.from(txn['group'], 'base64');
+///
+// Takes an unsigned decoded transaction object and converts strings into Uint8Arrays
+// for note, appArgs, approval and close programs. Then returns a transactionBuilder encoded value
```

Comment:
this comment may be outdated?

Reply:
Yeah. I have updated it to more closely align to the new functionality.